server/server.go

```diff
@@ -150,6 +151,22 @@ func root(w http.ResponseWriter, r *http.Request) {
 	_, _ = w.Write([]byte("<html><body>Mattermost Push Proxy</body></html>"))
 }
 
+func (s *Server) version(w http.ResponseWriter, _ *http.Request) {
+	info := version.VersionInfo()
+	ret := map[string]interface{}{
+		"version": info.BuildVersion,
+		"hash":    info.BuildHash,
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(ret); err != nil {
```

Comment:
Any reason for not marshaling the info struct? We could always omit fields through struct tags if needed but at least we'd avoid the map altogether.

Reply:
I didn't want to publicly expose what would be considered internal details, and this is all we needed. Happy to marshal it all, but maybe I'll ask sec first?

Reply:
Just expose the ones we need for now and `json:"-"` the rest?

Reply:
Ah, right, missed that part of your response. :)